dolphinscheduler-spi/src/main/java/org/apache/dolphinscheduler/spi/plugin/DolphinPluginDiscovery.java

```diff
@@ -83,6 +83,7 @@ private DolphinPluginDiscovery() {
 
         return listClasses(file.toPath()).stream()
                 .filter(name -> classInterfaces(name, classLoader).contains(DolphinSchedulerPlugin.class.getName()))
+                .map(plugin -> plugin.replace(File.separatorChar, '.'))
```

Comment:
You don't have to make a change here, the main reason is the method `classInterfaces` didn't work well in Windows. 
It is better to fix the method 'DolphinPluginDiscovery.javaName'.
BTY, is the `binaryName` method exist problem in Windows? I am not sure.

Reply:
@ruanwenjun thanks, I submit it again. 
+ This class contains both conversations from '/' to '.' and from '\\' to '.'
+ it's a little tricky: classPaths(in `classInterfaces`) are platform-independent(always '/'), and  paths like filePath  are  platform-dependent(windows: '\\' unix-like:  '/'), so the` javaName` method would be better to have the ablity to deal with both '/' and '\\'.

Reply:
@kyoty Can we direct use `File.separator`?

Reply:
Thanks @ruanwenjun, It can't work, I've tried it yesterday.

Reply:
I will review later, some details need to be confirmed

Reply:
Yes, we need to confirm whether the `reader.getSuperName()` will return '\\'.
```java
/**
   * Returns the internal of name of the super class (see {@link Type#getInternalName()}). For
   * interfaces, the super class is {@link Object}.
   *
   * @return the internal name of the super class, or <tt>null</tt> for {@link Object} class.
   * @see ClassVisitor#visit(int, int, String, String, String, String[])
   */
  public String getSuperName() {
    // super_class is after the access_flags and this_class fields (2 bytes each).
    return readClass(header + 4, new char[maxStringLength]);
  }

/**
   * Returns the internal name of the class corresponding to this object or array type. The internal
   * name of a class is its fully qualified name (as returned by Class.getName(), where '.' are
   * replaced by '/'). This method should only be used for an object or array type.
   *
   * @return the internal name of the class corresponding to this object type.
   */
  public String getInternalName() {
    return valueBuffer.substring(valueBegin, valueEnd);
  }
```

Reply:
Sorry, I am wrong.
```java
return listClasses(file.toPath()).stream()
                .filter(name -> classInterfaces(name, classLoader).contains(DolphinSchedulerPlugin.class.getName()))
                .collect(Collectors.toSet());
```
Your implementation is right.

Reply:
Thanks @ruanwenjun, The implementation has been updated.

Reply:
@kyoty Sorry, I apologize for my previous review, I re-look at the code here.
The method `discoverPluginsFromArtifact` is aim to get the plugin class name.
In the below lines, there might two steps.
```java
return listClasses(file.toPath()).stream()
                .filter(name -> classInterfaces(name, classLoader).contains(DolphinSchedulerPlugin.class.getName()))
                .collect(Collectors.toSet());
```
First, `listClasses`, this method is aim to return the class in the file path. 
Second, `filter`, this method is aim to filter the plugin class.
And currently this cannot work well in windows.

In the `listClasses` method, it will use `javaName` to transform file name to class name, as discuss previous, this method cannot work well in windows. We need to modify the `javaName` method.
![image](https://user-images.githubusercontent.com/22415594/123602099-fbd1f880-d82a-11eb-9bc1-eb21053d53f2.png)

When you modify the method `javaName`, it also doesn't work in windows, because the `classInterfaces` also use the `javaName` method. The problem is that the `reader.getSuperName()` is always return 'xx/xx/xx', independent of OS file splitter.
![image](https://user-images.githubusercontent.com/22415594/123602555-70a53280-d82b-11eb-8f19-1793aece9e21.png)

I am not sure if I am expressing myself clearly.

Reply:
@ruanwenjun  Thanks for your patient view. 
```java
return listClasses(file.toPath()).stream()
                .filter(name -> classInterfaces(name, classLoader).contains(DolphinSchedulerPlugin.class.getName()))
                .collect(Collectors.toSet());
```
as the listClasses determines what kind of str to return. I try to add a new method in replace javaName in listClasses, now it works fine in windows, I'll test it in unix system tomorrow,  do you have any better idea?

Reply:
@ruanwenjun  I've test it under ubuntu, it works as expected, thanks for your great help.